### PR TITLE
[inductor] set sanitize_overflow=False for triton kernels

### LINKED ISSUE #139348: inductor cumsum w/ upstream triton fails accuracy and device asserts
### 🐛 Describe the bug

Repro:
```python
import torch

def fn(x):
    return x.cumsum(0)

x = torch.rand(100, 4000, device='cuda')

breakpoint()
expect = fn(x)
actual = torch.compile(fn)(x)

torch.testing.assert_allclose(expect, actual)
```

### Versions

5c6d35482e7cb3d22f9ab8cef37fa16c05a24644 pytorch, 018c139d2b843c29c3b4d4d2e6f3e672b8ff0b3a triton. H100.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @bertmaher @int3 @nmacchioni @embg @peterbell10

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139502

In upstream triton, https://github.com/triton-lang/triton/pull/4589 introduces overflow checks. However, overflow checks likely add some overhead, and have some correctness bugs at the moment (e.g. https://github.com/triton-lang/triton/pull/5033). Let's set `sanitize_overflow=False` but keep `debug=True` so that we can keep using device_assert but without the additional asserts added by `sanitize_overflow`.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by bertmaher — APPROVED** [review]

File: torch/_inductor/runtime/triton_heuristics.py [hunk 1]
```diff
@@ -442,6 +442,7 @@ def _precompile_config(self, cfg: Config, warm_cache_only: bool):
                 "num_warps": compile_meta["num_warps"],
                 "num_stages": compile_meta["num_stages"],
                 "debug": compile_meta["debug"],
+                "sanitize_overflow": False,  # turn off additional asserts added for overflow checks
```
[the remark this answers is not in the document]
Comment (davidberard98) [reply torch/_inductor/runtime/triton_heuristics.py:445]: I didn't look too closely but:
* relying on PyTorch CI to catch it if not.
* I did test with AMD w/ new version of triton, which doesn't have `sanitize_overflow` defined in HIPOptions, and it doesn't fail.
